dbms/src/Functions/FunctionsArithmetic.h

```diff
@@ -264,9 +348,19 @@ struct BitShiftLeftImpl
     template <typename Result = ResultType>
     static inline Result apply(A a, B b)
     {
-        return static_cast<Result>(a)
-            << static_cast<Result>(b);
+        return static_cast<Result>(a) << static_cast<Result>(b);
+    }
+
+#if USE_EMBEDDED_COMPILER
+    static constexpr bool compilable = true;
+
+    static inline llvm::Value * compile(llvm::IRBuilder<> & b, llvm::Value * left, llvm::Value * right, bool)
+    {
+        if (!left->getType()->isIntegerTy())
+            throw Exception("BitShiftLeftImpl expected an integral type", ErrorCodes::LOGICAL_ERROR);
```

Comment:
It doesn't. Example:

```
SELECT bitShiftLeft(1.1, number) FROM system.numbers LIMIT 10
```

Reply:
That simply casts 1.1 to an integer type, which is done by `FunctionBinaryArithmetic::compileImpl`. *This* method should never receive a non-int type.